esx_service/utils/vmdk_utils.py

```diff
@@ -68,15 +70,34 @@ def get_vmdk_path(path, vol_name):
     return os.path.join(path, "{0}.vmdk".format(vol_name))
 
 
-def list_vmdks(path):
-    """ Return a list all VMDKs in a given path """
-    try:
-        return [f for f in os.listdir(path)
-                if vmdk_is_a_descriptor(os.path.join(path, f))]
-    except OSError as e:
-        # dockvols may not exists on a datastore, so skip it
+def list_vmdks(path, volname="", show_snapshots=False):
+    """ Return a list of VMDKs in a given path. Filters out non-descriptor
+    files.
+
+    Params:
+    path -  where the vmdks are looked for
+    volname - if passed, only files related to this vmdks will be returned. Usefule when
+            doing volume snapshot inspect
+    show_snapshots - if set to False, all VMDKS (including delta files) will be returned
```

Comment:
It looks in the code, and from the name that all snapshots are returned if show_snapshots is `True`